### ISSUE TITLE
Throw in hole: stacks notes aren't rendering cards

### ISSUE THREAD

**Issue by DarthGandalf** [Hanabi-Live/hanabi-live#2875]
**Describe the bug**
It's possible to make notes on stacks, but they no longer change appearance of the stacks - it always looks empty

**To Reproduce**

1. start TIIAH game
2. add a note on a stack
4. See nothing changed

**Expected behavior**

the stack appearance should follow the note

**Screenshots**
![Выделение_474](https://github.com/Hanabi-Live/hanabi-live/assets/325092/3786d703-0376-475b-a237-470c743ed81f)

**Additional context**
Can't share replay, because during replay it doesn't hide the cards.

**Comment by newduke** [comment]
Verified broken

**Comment by NZadeh** [comment]
Based on the timing of this report and some of the code history, I suspect one of these may be to blame:

- "refactor: SuitIndex and Rank types used for stronger typing"
    - https://github.com/Hanabi-Live/hanabi-live/commit/bd7457fb78b185ef70562c7e82c8ec9d46056107?diff=split&w=0#diff-57c4aa25ee57873f21076408fce46b29bebb0afc29c016bc4d4774a6c75151b2
- "refactor: remove special ranks from codebase" 
    - https://github.com/Hanabi-Live/hanabi-live/commit/c279b89842ce605dd9d348c9d269f5d3b7dbeca3
- "fix: stack base pips"
    - https://github.com/Hanabi-Live/hanabi-live/commit/318a99f3311a39cc29a6caf0e06f27c851ea15fb

@Zamiell do you think anything about how `isStackBase` is set, compared to the old methodology, could be affecting the code here: https://github.com/Hanabi-Live/hanabi-live/blame/main/packages/client/src/game/ui/HanabiCard.ts#L450 ?  (I might do a bit more code digging myself, but you will probably know immediately once you look at it :D)

**Comment by Zamiell** [comment]
i have no idea what the problem is, if you want to investigate you can just start adding debug statements to see what is going on

**Comment by NZadeh** [comment]
Okay, I figured out the problem...

https://github.com/Hanabi-Live/hanabi-live/commit/318a99f3311a39cc29a6caf0e06f27c851ea15fb replaced 
```
return `card-${suitName}-${rankToShow}`;
```
with
```
const bareNameRank = this.isStackBase
      ? STACK_BASE_RANK
      : rankToShow ?? UNKNOWN_CARD_RANK;
return `card-${bareNameSuit}-${bareNameRank}`;
```

Note that this makes it impossible for the card's bareName to _ever_ change from e.g. "card-Teal-StackBase". Whereas before, it was able to be overwritten by a note (that set `rankToShow`) to e.g. "card-Teal-3".

If I comment out:
```
  getBareNameRank(rankToShow: Rank | null): string {
    // if (this.isStackBase) {
    //   return CARD_IMAGE_STACK_BASE_RANK_NAME;
    // }
   ...
 } 
```
we see that the note actually works:

![image](https://github.com/Hanabi-Live/hanabi-live/assets/4044138/c9db7683-2ae4-48bc-8ee8-3cd227ab4970)

So the trick is to return the code to a state where the "rankToShow" can override the logic for `isStackBase`... :)

**Comment by NZadeh** [comment]
Well, I should be sleeping, but instead I'm improving Throw It in a Hole for everyone... 😢 
Please take a look at https://github.com/Hanabi-Live/hanabi-live/pull/2928 as it should fix this issue.

**Comment by NZadeh** [comment]
This issue should be fixed now, and is fixed when I test on hanab.live. Thanks for the help James! You should be able to mark as closed. I don't think I attached the PR correctly to this issue... :'(

**Comment by Zamiell** [comment]
yeah next time you can do "fixes #2875" in the PR title and it will close the issue automatically

**Comment by Zamiell** [comment]
err i mean in the PR description
